packages/js/data/src/products/resolvers.ts

```diff
@@ -48,7 +49,9 @@ export function* getProducts( query: Partial< ProductQuery > ) {
 export function* getProduct( productId: number ) {
 	try {
 		const product: Product = yield apiFetch( {
-			path: `${ WC_PRODUCT_NAMESPACE }/${ productId }`,
+			path: addQueryArgs( `${ WC_PRODUCT_NAMESPACE }/${ productId }`, {
+				context: 'edit',
```

Comment:
`context` is important for the returned data type as the default `view` context formats returned data in a way that is not ideal for editing.

Allowing `context` to be changed would better, but I think we can forego that since this data store will most likely be replaced with the CRUD data store generator in the near future.